src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -3567,6 +3559,11 @@ internal TElement[] ToArray()
         }
     }
 
+    // NOTE: DO NOT DELETE THE FOLLOWING DEBUG VIEW TYPES.
+    // Although it might be tempting due to them not be referenced anywhere in this library,
+    // Visual Studio currently depends on their existence to enable the "Results" view in 
+    // watch windows.
+
```

Comment:
I do not see why it simply does not do _enumerable.ToArray() 
Also _cachedCollection seems to be there to cache the values while it is not used. 
The ```_cachedCollection = null;``` in SystemCore_EnumerableDebugView ctor seems bogus.

The code clearly could benefit from some clean up. My guess - the scary comment makes people look the other way.